cpp/oneapi/dal/backend/common.hpp

```diff
@@ -63,6 +63,16 @@ inline constexpr Integer up_multiple(Integer x, Integer multiple) {
     return y + z;
 }
 
+/// Checks if 'x' is power of 2.
+/// Example: down_pow2(10) == false
+/// Example: down_pow2(16) == true
+template <typename Integer>
+inline constexpr bool is_pow2(Integer x) {
+    static_assert(std::is_integral_v<Integer>);
+    ONEDAL_ASSERT(x > 0);
```

Comment:
Probably it's better to return `false` if `x <= 0` in this function rather than putting an assert